packages/bot-web-ui/src/stores/app-store.js

```diff
@@ -25,13 +25,100 @@ export default class AppStore {
         this.api_helpers_store = null;
     }
 
+    getErrorForNonEuClients = () => ({
+        text: localize(
+            'Unfortunately, this trading platform is not available for EU Deriv account. Please switch to a non-EU account to continue trading.'
+        ),
+        title: localize('Deriv Bot is unavailable for this account'),
+        link: localize('Switch to another account'),
+    });
+
+    getErrorForEuClients = (is_logged_in = false) => {
+        return {
+            text: ' ',
+            title: is_logged_in
+                ? localize('Deriv Bot is not available for EU clients')
+                : localize('Deriv Bot is unavailable in the EU'),
+            link: is_logged_in && localize("Back to Trader's Hub"),
+            route: routes.traders_hub,
+        };
+    };
+
+    handleErrorForEu = (show_default_error = false) => {
+        const { client, common, ui, traders_hub } = this.core;
+        const toggleAccountsDialog = ui?.toggleAccountsDialog;
+        const recovery_low_risk_cr_eu = traders_hub?.selected_region;
+
+        if (!client?.is_logged_in && client?.is_eu_country) {
+            return showDigitalOptionsUnavailableError(common.showError, this.getErrorForEuClients());
+        }
+
+        if (!client.is_landing_company_loaded) {
+            return false;
+        }
+
+        if (
+            window.location.pathname === routes.bot &&
+            client.should_show_eu_error &&
+            (traders_hub.content_flag === ContentFlag.EU_REAL || traders_hub.content_flag === ContentFlag.EU_DEMO) &&
```

Comment:
This condition is always false because:
when the`content_flag === ContentFlag.EU_REAL || traders_hub.content_flag === ContentFlag.EU_DEMO` it means the user is EU and we just shoe the regulation switcher when user is `LOW_RISK_CR_NON_EU` or `LOW_RISK_CR_EU`

Reply:
Actually, this is an expression of the `true` for the case when it is EU only account
For both cases such as:
 1. EU only account➝ Switch to Demo Account ➝ Go to Dbot page
 2. EU only account➝ Switch to Real Account ➝ Go to Dbot page

Reply:
But i've removed condition `content_flag === ContentFlag.EU_REAL || traders_hub.content_flag === ContentFlag.EU_DEMO` since `traders_hub.content_flag` is guaranteed to be calculated correctly in async mode with the applied check  `client.is_landing_company_loaded` in the start. We really don't need additional variables anymore such as `recovery_low_risk_cr_eu`, it was removed too.